includes/Classifai/Providers/Azure/Language.php

```diff
@@ -0,0 +1,391 @@
+					],
+				],
+			],
+			'tasks'         => [
+				[
+					'kind'       => 'AbstractiveSummarization',
+					'taskName'   => 'Classifai Summarization ' . $post_id,
+					'parameters' => [
+						/**
+						 * Filter the summary length.
+						 * Possible values are 'oneSentence', 'short', 'medium', 'long'.
+						 * Default is 'oneSentence'.
+						 *
+						 * @since 3.2.0
+						 * @hook classifai_azure_language_summary_length
+						 * @param string $summary_length The summary length.
+						 * @return string
+						 */
+						'summaryLength' => apply_filters( 'classifai_azure_language_summary_length', 'oneSentence' ),
+					],
+				],
+			],
+		];
+
+		$request = wp_remote_post(
+			$endpoint_url,
+			[
+				'headers' => [
+					'Ocp-Apim-Subscription-Key' => $api_key,
+					'Content-Type'              => 'application/json',
+				],
+				'body'    => wp_json_encode( $body ),
+			]
+		);
+
+		$headers = wp_remote_retrieve_headers( $request );
+
+		if ( ! is_wp_error( $request ) ) {
+			$response = json_decode( wp_remote_retrieve_body( $request ) );
+			if ( ! empty( $response->error ) ) {
+				return new WP_Error( 'summary_request', $response->error->message );
+			} elseif ( empty( $headers['operation-location'] ) ) {
+				return new WP_Error( 'summary_request', esc_html__( 'There was an error requesting the summary.', 'classifai' ) );
+			} else {
+				return $headers['operation-location'];
+			}
+		}
+
+		return $request;
+	}
+
+	/**
+	 * Get the text analysis job response
+	 *
+	 * The endpoint URL is returned in the 'operation-location' header of the initial request.
+	 * The job response will return a 'status' property if the job is completed.
+	 * If the job is not completed, wait a second and try again.
+	 *
+	 * @param string $url The URL to analyze.
+	 * @see https://learn.microsoft.com/en-us/azure/ai-services/language-service/summarization/quickstart?tabs=text-summarization%2Cmacos&pivots=rest-api
+	 * @return mixed
+	 */
+	private function retrieve_summary( $url ) {
+		$api_key = $this->feature_instance->get_settings( static::ID )['api_key'];
+
+		$request = safe_wp_remote_get(
+			$url,
+			[
+				'headers' => [
+					'Ocp-Apim-Subscription-Key' => $api_key,
+					'Content-Type'              => 'application/json',
+				],
+			]
+		);
+
+		$summary = '';
+
+		if ( ! is_wp_error( $request ) ) {
+			$response = json_decode( wp_remote_retrieve_body( $request ) );
+			if ( ! empty( $response->error ) ) {
+				return new WP_Error( 'auth', $response->error->message );
+			}
+
+			while ( 'succeeded' !== $response->status ) {
+				sleep( .5 );
+				$request  = safe_wp_remote_get(
+					$url,
+					[
+						'headers' => [
+							'Ocp-Apim-Subscription-Key' => $api_key,
+							'Content-Type'              => 'application/json',
+						],
+					]
+				);
+				$response = json_decode( wp_remote_retrieve_body( $request ) );
+			}
+
+			$summary = $response->tasks->items[0]->results->documents[0]->summaries[0]->text;
```

Comment:
Just want to confirm that the array indexes at `[0]` will always contain those properties, or if we should check via `isset()` and `count() > 0`?